django_log_formatter_asim/__init__.py

```diff
@@ -1,177 +1,93 @@
+    def get_log_dict(self):
+        return self._get_log_dict_base()
 
 
 class ASIMRequestFormatter(ASIMFormatterBase):
-    def get_event(self):
-        zipkin_headers = getattr(
-            settings,
-            "DLFE_ZIPKIN_HEADERS",
-            ("X-B3-TraceId", "X-B3-SpanId"),
-        )
-
-        extra_labels = {}
-
-        for zipkin_header in zipkin_headers:
-            if getattr(
-                self.record.request.headers,
-                zipkin_header,
-                None,
-            ):
-                extra_labels[zipkin_header] = self.record.request.headers[
-                    zipkin_header
-                ]  # noqa E501
+    def get_log_dict(self):
+        log_dict = self._get_log_dict_base()
+
+        request = self.record.request
+
+        # Source fields...
+        log_dict["SrcIpAddr"] = request.environ.get("REMOTE_ADDR", None)
+        log_dict["IpAddr"] = log_dict["SrcIpAddr"]
+        log_dict["SrcPortNumber"] = request.environ.get("SERVER_PORT", None)
+
+        # Acting Application fields...
+        http_user_agent = getattr(request, "user_agent", None)
+        if not http_user_agent:
+            http_user_agent = getattr(request.headers, "user_agent", None)
+        if not http_user_agent:
+            http_user_agent = request.META.get("HTTP_USER_AGENT", None)
+        log_dict["HttpUserAgent"] = http_user_agent
+
+        user = getattr(request, "user", None)
+        user_id = None
+        username = None
+        if user:
+            user_id = getattr(user, "id", None)
+            if getattr(settings, "DLFA_LOG_SENSITIVE_USER_DATA", False):
+                username = getattr(user, "username", None)
+                if not username:
+                    username = getattr(user, "email", None)
+            else:
+                username = "REDACTED"
+        log_dict["SrcUserId"] = user_id
+        log_dict["SrcUsername"] = username
 
-        logger_event = self._get_event_base(
-            extra_labels=extra_labels,
-        )
+        return log_dict
 
-        parsed_url = urlparse(self.record.request.build_absolute_uri())
-
-        ip = self._get_ip_address(self.record.request)
-
-        request_bytes = len(self.record.request.body)
-
-        logger_event.url(
-            path=parsed_url.path,
-            domain=parsed_url.hostname,
-        ).source(
-            ip=self._get_ip_address(self.record.request)
-        ).http_response(status_code=getattr(self.record, "status_code", None)).client(
-            address=ip,
-            bytes=request_bytes,
-            domain=parsed_url.hostname,
-            ip=ip,
-            port=parsed_url.port,
-        ).http_request(
-            body_bytes=request_bytes,
-            body_content=self.record.request.body,
-            method=self.record.request.method,
-        )
-
-        user_agent_string = getattr(
+    def _get_user_agent(self):
+        return getattr(
             self.record.request.headers,
```

Comment:
Can we assume that we have a request?  🤔  Not all logging is done in the context of a request object.

Reply:
What are the other options and do we need to accommodate them in this first iteration (MVP)?

Reply:
This method is not being used anymore, but I'll pull the user agent stuff into it and use it to better encapsulate that logic.